test/cli/samples/watch/watch-event-hooks/_config.js

```diff
@@ -0,0 +1,24 @@
+const { assertIncludes } = require('../../../../utils.js');
+
+module.exports = {
+	description: 'onStart event hoot shell command executes correctly',
+	command:
+		'rollup -cw --watch --watch.onStart "echo start" --watch.onBundleStart "echo bundleStart" --watch.onBundleEnd "echo bundleEnd" --watch.onEnd "echo onEnd" --watch.onError "echo onError"',
```

Comment:
`-cw` includes `--watch` 😉

Reply:
lol oops.

I could use some guidance on how to do this test exactly. The `test/cli/samples/` suite of test all seem to do tests on either console output or bundle file output. the one I threw up there was just a POC to test all of the watch event hooks at once by just looking for the echos being written

I can get that test working or I can take another direction. Would you like me to break up each hook into its own test? should I test that `--silent` doesn't print to stderr?

Reply:
No, I think the test is fine in principle, and testing all hooks together seems adequate to me considering me that there is not too much logic involved. You should just make sure to make it a little more robust, e.g. see my other suggestion of preprocessing the output by splitting it into lines, doing some mapping and filtering and possibly doing an `deepStrictEqual` on the result. Looks to me like "echo" should also work on Windows, which is definitely a bonus.